ssl/statem/extensions_clnt.c

```diff
@@ -17,12 +17,10 @@ EXT_RETURN tls_construct_ctos_renegotiate(SSL_CONNECTION *s, WPACKET *pkt,
                                           size_t chainidx)
 {
     if (!s->renegotiate) {
+        /**
+         * TODO(DTLSv1.3): Fix this when resumption and renegotiation works
+         */
```

Comment:
Isn't it just this?:

```suggestion
        if (SSL_CONNECTION_IS_DTLS(s)
                && ssl_version_cmp(s->min_proto_version, DTLS1_3_VERSION) >= 0)
            return EXT_RETURN_NOT_SENT;
```

The renegotiation extension is not relevant for DTLSv1.3 since it doesn't support renegotiation.

Reply:
I thought so too. But then I get failures like:
```
    # 0041575DF87F0000:error:0A000152:SSL routines:final_renegotiate:unsafe legacy renegotiation disabled:../ssl/statem/extensions.c:946:
```

I haven't spent time looking at what goes wrong yet.

Reply:
Yes that seems reasonable - although shouldn't it be:

```` C
    if (!ossl_assert(versiona > 0) || !ossl_assert(versionb > 0))
        return 0;
````

Reply:
Good suggestion!

I think I want to do it as a seperate PR since we will need to fix checks in other places. Is there a way to get a full stack dump when ossl_assert is hit?

Reply:
Not as far as I know. I usually just put it in the debugger.

Reply:
I've fixed the behaviour of ssl_version_cmp in
https://github.com/openssl/openssl/pull/24293 (feature/dtls-1.3)
https://github.com/openssl/openssl/pull/24292 (master)

Please have a look. I'd like to rebase this branch on those changes.